components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/MetadataNodeEntityAdapter.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+
+package org.sonatype.nexus.repository.storage;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+
+import org.sonatype.nexus.common.collect.NestedAttributesMap;
+import org.sonatype.nexus.common.entity.EntityId;
+import org.sonatype.nexus.orient.entity.CollectionEntityAdapter;
+import org.sonatype.nexus.repository.Repository;
+
+import com.google.common.base.Function;
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx;
+import com.orientechnologies.orient.core.id.ORID;
+import com.orientechnologies.orient.core.metadata.schema.OClass;
+import com.orientechnologies.orient.core.metadata.schema.OClass.INDEX_TYPE;
+import com.orientechnologies.orient.core.metadata.schema.OType;
+import com.orientechnologies.orient.core.record.impl.ODocument;
+import com.orientechnologies.orient.core.sql.OCommandSQL;
+import org.joda.time.DateTime;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_ATTRIBUTES;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_BUCKET;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_FORMAT;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_LAST_UPDATED;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_REPOSITORY_NAME;
+
+/**
+ * {@link MetadataNode} entity-adapter.
+ *
+ * @since 3.0
+ */
+public abstract class MetadataNodeEntityAdapter<T extends MetadataNode>
+    extends CollectionEntityAdapter<T>
+{
+
+  private final BucketEntityAdapter bucketEntityAdapter;
+
+  @Inject
```

Comment:
It looks odd, especially since the first parameter is an unqualified String

(FWIW it doesn't make sense to use `@Inject` on abstract class constructors since they can't be instantiated as-is)